drivers/gpio/gpio_nrfx.c

```diff
@@ -368,18 +422,59 @@ static uint32_t check_level_trigger_pins(const struct device *port)
 	uint32_t pin = 0U;
 	uint32_t bit = 1U << pin;
 
-	while (level_pins) {
-		if (level_pins & bit) {
+	uint32_t port_latch = 0;
+
+#if IS_ENABLED(CONFIG_GPIO_NRF_INT_EDGE_USING_SENSE)
```

Comment:
Defining a macro that looks like a variable but aliases one of two variables is too complex.  How about:
```
uint32_t port_latch = 0;
uint32_t check_pins = level_pins;

if (IS_ENABLED(CONFIG_GPIO_NRF_INT_EDGE_USING_SENSE)) {
  check_pins = data->pin_int_en;
  /* Read LATCH, which will tell us which pin has changed its state. */
  port_latch = cfg->port->LATCH;
}
```

Reply:
We can't use `cfg->port->LATCH;` under `if (IS_ENABLED(...))`, as it won't compile for MCUs without `LATCH` register.

But we can still do:
```
uint32_t check_pins = level_pins;

if (IS_ENABLED(CONFIG_GPIO_NRF_INT_EDGE_USING_SENSE)) {
  check_pins = data->pin_int_en;
}
```
it will produce slightly bigger code (16B for my builds) and I guess a bit more stack usage (there will be two separate `level_pins` and `check_pins` variables) in case of `CONFIG_GPIO_NRF_INT_EDGE_USING_GPIOTE=y`. The idea behind using check_pins macro was to prevent overhead when GPIOTE is used. If you feel that this is fine, that I will make `check_pins` a C variable, as you suggested.

Reply:
Sorry, you're right about `LATCH`.  Annoying but unavoidable.

I would expect the compiler to optimize away any redundant variables.  Fix it any way you like that doesn't involve `#define`ing an alias that expands to a declared variable.